src/gui/widgets/subtitlelistwidget.cpp

```diff
@@ -297,6 +310,8 @@ void SubtitleListWidget::handleTracklistChange(
         m_primaryLock.unlock();
         m_subRegexLock.unlock();
 
+        m_didParsePrimarySubs = m_subtitleMap[primarySid].length() > 0;
```

Comment:
This is still bad. You should save these values in another map or change the type of the subtitle map to a pair containing a boolean and a list. These should also be pointers to that value since it may change.

Reply:
I would really appreciate if you could provide a sniper here because I clearly don't know about the best practices.

Reply:
Add this
```cpp
QHash<int64_t, bool> m_subtitleParsed;
```
Change these types to:
```cpp
bool *m_didParsePrimarySubs = nullptr;
bool *m_didParseSecondarySubs = nullptr;
```

And then fill in the rest, setting `m_subtitleParsed` as appropriate and clearing it in `clearCachedSubtitles()`.